src/psyclone/tests/psyir/nodes/omp_directives_test.py

```diff
@@ -471,6 +472,88 @@ def test_omp_master_nested_validate_global_constraints(monkeypatch):
             "region") in str(excinfo.value)
 
 
+def test_omptaskwait_dag_name():
+    '''Test the omp_taskwait dag_name method'''
+    _, invoke_info = parse(os.path.join(BASE_PATH, "1_single_invoke.f90"),
+                           api="dynamo0.3")
+    psy = PSyFactory("dynamo0.3", distributed_memory=False).\
+        create(invoke_info)
+    schedule = psy.invokes.invoke_list[0].schedule
+    taskwait = OMPTaskwaitDirective()
+    schedule.addchild(taskwait, 0)
+    assert taskwait.dag_name == "OMP_taskwait_1"
+
+
+def test_omptaskwait_strings():
+    ''' Test the begin_string and end_string methods of
+        the OMPTaskwait directive '''
+    taskwait = OMPTaskwaitDirective()
+
+    assert taskwait.begin_string() == "omp taskwait"
+    assert taskwait.end_string() == ""
+
+
+def test_omptaskwait_node_str():
+    '''Test the omp_taskwait node_str method'''
+    _, invoke_info = parse(os.path.join(BASE_PATH, "1_single_invoke.f90"),
+                           api="dynamo0.3")
+    psy = PSyFactory("dynamo0.3", distributed_memory=False).\
+        create(invoke_info)
+    schedule = psy.invokes.invoke_list[0].schedule
+    taskwait = OMPTaskwaitDirective()
+    schedule.addchild(taskwait, 0)
+    directive = colored("Directive", Directive._colour)
+    expected_output = directive + "[OMP taskwait]"
+    assert taskwait.node_str() == expected_output
+
+
+def test_omptaskwait_gencode():
+    '''Check that the gen_code method in the OMPTaskwaitDirective
+    class generates the expected code. Use the gocean API.
+    '''
+    _, invoke_info = parse(os.path.join(GOCEAN_BASE_PATH, "single_invoke.f90"),
+                           api="gocean1.0")
+    taskwait = OMPTaskwaitDirective()
+    parallel = OMPParallelTrans()
+    master = OMPMasterTrans()
+    psy = PSyFactory("gocean1.0", distributed_memory=False).\
+        create(invoke_info)
+    schedule = psy.invokes.invoke_list[0].schedule
+    taskloop = OMPTaskloopTrans()
+
+    schedule.addchild(taskwait, 1)
+    taskloop.apply(schedule.children[0])
+    master.apply([schedule.children[0], schedule.children[1]])
+    parallel.apply(schedule.children[0])
+    goceantrans = GOceanExtractTrans()
```

Comment:
The test fails without it so I think so.

Reply:
Oh yes, now that I take a proper look this test is there because things like ExtractNode still use the old gen_code way of creating code rather than using a PSyIR backend.